velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -29,17 +29,39 @@ SubstraitVeloxExprConverter::toVeloxExpr(
         kDirectReference: {
       const auto& directRef = substraitField.direct_reference();
       int32_t colIdx = substraitParser_.parseReferenceSegment(directRef);
+
+      const auto& inputTypes = inputType->children();
       const auto& inputNames = inputType->names();
       const int64_t inputSize = inputNames.size();
-      if (colIdx <= inputSize) {
-        const auto& inputTypes = inputType->children();
-        // Convert type to row.
+      VELOX_CHECK_LT(
+          colIdx, inputSize, "Missing the column with id '{}' .", colIdx);
+      std::optional<int32_t> childIdx;
+      if (directRef.struct_field().has_child()) {
+        childIdx = substraitParser_.parseReferenceSegment(
+            directRef.struct_field().child());
+      }
+
+      if (!childIdx.has_value()) {
         return std::make_shared<core::FieldAccessTypedExpr>(
             inputTypes[colIdx],
             std::make_shared<core::InputTypedExpr>(inputTypes[colIdx]),
             inputNames[colIdx]);
+      }
+
+      // Select a subfield in a struct by name.
```

Comment:
I'm not sure how this works. Would you explain? Does this support multiply-nested structs, e.g. a.b.c.d ? Would you add a test?

Reply:
I add a test case for multiply-nested structs. I use two row_constructors to create a type like a.b.c. And test passed.